File: lib/hurray-ruby-sdk/hrect.rb
class HRect
  
  attr_accessor :x, :y, :width, :height
  
  def initialize(x, y, w, h)
    @x = x
    @y = y
    @width = w
    @height = h
  end
  
  def self.create(options = {:x => 0, :y => 0, :width => 0, :height => 0})
    HRect.new(options[:x], options[:y], options[:width], options[:height])
  end
  
end

File: lib/hurray-objc-sdk/HRect.m
//
//  HRect.m
//  hurray-objc-sdk
//
//  Created by Paolo Castro on 3/7/11.
//  Copyright 2011 Jalasoft. All rights reserved.
//

#import "HRect.h"


@implementation HRect

@synthesize x, y, width, height;

- (id) initialize:(int)x :(int)y :(int)width :(int)height
{
	self = [self init];
	self.x = x;
	self.y = y;
	self.width = width;
	self.height = height;
	return self;
}

- (CGRect) toCGRect
{
	CGRect rect;
	rect.origin.x = x;
	rect.origin.y = y;
	rect.size.width = width;
	rect.size.height = height;
	return rect;
}

@end
